vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt

```diff
@@ -0,0 +1,317 @@
+    ${lpc54018_dir}/drivers/fsl_spi.c
+    ${lpc54018_dir}/drivers/fsl_spi_dma.c
+    ${lpc54018_dir}/drivers/fsl_spifi.c
+    ${lpc54018_dir}/drivers/fsl_spifi_dma.c
+    ${lpc54018_dir}/drivers/fsl_usart.c
+    ${lpc54018_dir}/drivers/mflash_drv.c
+    ${lpc54018_dir}/drivers/mflash_file.c
+)
+
+set(lpc54018_includes
+    "${lpc54018_dir}"
+    "${lpc54018_dir}/drivers"
+    "${lpc54018_dir}/cmsis_drivers"
+    "${lpc54018_dir}/middleware/wifi_qca"
+    "${lpc54018_dir}/middleware/wifi_qca/common_src/hcd"
+    "${lpc54018_dir}/middleware/wifi_qca/common_src/include"
+    "${lpc54018_dir}/middleware/wifi_qca/common_src/stack_common"
+    "${lpc54018_dir}/middleware/wifi_qca/common_src/wmi"
+    "${lpc54018_dir}/middleware/wifi_qca/custom_src/include"
+    "${lpc54018_dir}/middleware/wifi_qca/custom_src/stack_custom"
+    "${lpc54018_dir}/middleware/wifi_qca/include"
+    "${lpc54018_dir}/middleware/wifi_qca/include/AR6002"
+    "${lpc54018_dir}/middleware/wifi_qca/include/AR6002/hw2.0/hw"
+    "${lpc54018_dir}/middleware/wifi_qca/port"
+    "${lpc54018_dir}/middleware/wifi_qca/port/boards/lpc54018iotmodule/freertos"
+    "${lpc54018_dir}/middleware/wifi_qca/port/boards/lpc54018iotmodule/freertos/gt202"
+    "${lpc54018_dir}/middleware/wifi_qca/port/drivers/flexcomm_freertos"
+    "${lpc54018_dir}/middleware/wifi_qca/port/env/freertos"
+    "${lpc54018_dir}/middleware/usb/device"
+    "${lpc54018_dir}/middleware/usb/include"
+    "${lpc54018_dir}/middleware/usb/osa"
+    "${lpc54018_dir}/utilities"
+    "${lpc54018_dir}/utilities/io"
+    "${lpc54018_dir}/utilities/log"
+    "${lpc54018_dir}/utilities/str"
+    "${lpc54018_dir}/CMSIS/Include"
+)
+
+# Kernel
+afr_mcu_port(kernel)
+
+target_sources(
+    AFR::kernel::mcu_port
+    INTERFACE
+        ${lpc54018_src}
+        ${drivers_src}
+        ${mcuxpresso_src}
+        ${usb_device_src}
+        ${usb_osa_src}
+        ${wifi_qca_src}
+        ${utilities_src}
+        "${AFR_KERNEL_DIR}/portable/GCC/ARM_CM4F/port.c"
+        "${AFR_KERNEL_DIR}/portable/GCC/ARM_CM4F/portmacro.h"
+        "${AFR_KERNEL_DIR}/portable/MemMang/heap_4.c"
+)
+
+set(kernel_inc_dirs
+    ${lpc54018_includes}
+    "${AFR_KERNEL_DIR}/portable/GCC/ARM_CM4F"
+    "${AFR_KERNEL_DIR}/portable/CCS/ARM_CM3"
+    "${lpc54018_app_dir}/config_files"
+    "${lpc54018_app_dir}/application_code/nxp_code"
+    "$<IF:${AFR_IS_TESTING},${AFR_TESTS_DIR},${AFR_DEMOS_DIR}>/include"
+
+)
+
+target_include_directories(
+    AFR::kernel::mcu_port
+    INTERFACE $<$<NOT:$<COMPILE_LANGUAGE:ASM>>:${kernel_inc_dirs}>
+)
+
+
+# WiFi
+afr_mcu_port(wifi)
+target_sources(
+    AFR::wifi::mcu_port
+    INTERFACE "${lpc54018_ports_dir}/wifi/aws_wifi.c"
+)
+
+# PKCS11
+afr_mcu_port(pkcs11)
+target_sources(
+    AFR::pkcs11::mcu_port
+    INTERFACE 
+        "${lpc54018_ports_dir}/pkcs11/aws_pkcs11_pal.c"
+        "${lpc54018_ports_dir}/pkcs11/hw_poll.c"
+        "${AFR_MODULES_PORTS_DIR}/pkcs11/mbedtls/aws_pkcs11_mbedtls.c"
```

Comment:
We already defined a target that uses this default implementation, so you don't need to specify it here. Instead, use
```cmake
target_link_libraries(
    AFR::pkcs11::mcu_port
    INTERFACE pkcs11_mbedtls
)

Reply:
You mean "AFR::pkcs11_mbedtls" right? OK